### ISSUE TITLE
False positive MA0150 on anonymous type

### ISSUE THREAD

**Issue by jafaber** [meziantou/Meziantou.Analyzer#674]
### Version of the Meziantou.Analyzer NuGet package

2.0.132

### Rule Identifier

MA0150

### Target Framework

net7.0

### C# Language version

C# 12

### Description

Anonymous types have a "useful" ToString implementation, so they should be excluded from triggering this rule. Both when calling ToString directly as well as inside an interpolated string.

### Reproduction Steps

Minimal code:

```c#
    bool blah = true;
    var x = $"{blah} {new { blah }}";
    var y = new { blah }.ToString();
```

**Comment by meziantou** [comment]
The latest version should handle anonymous types. It should be available in a few minutes on nuget.org.

**Comment by jafaber** [comment]
Thanks, but it's not entirely fixed yet. The following still triggers a false positive:

`if (!Path.Exists(path)) throw new DirectoryNotFoundException(new { path }.ToString());`

Minimally:
```
        bool blah = true;
        var y = new { blah }.ToString();
```

**Comment by meziantou** [comment]
The new pacakge containing the fix will be available when https://github.com/meziantou/Meziantou.Analyzer/actions/runs/7423504595 completes
